src/app/DataAccessLayer/DataAccessLayerExtension.cs

```diff
@@ -19,15 +20,32 @@ public static class DataAccessLayerExtension
         /// <param name="cosmosKey">Cosmos Key</param>
         /// <param name="cosmosDatabase">Cosmos Database</param>
         /// <param name="cosmosCollection">Cosmos Collection</param>
+        /// <param name="timeout">Cosmos client timeout</param>
+        /// <param name="retry">Cosmos client retry</param>
         /// <returns>ServiceCollection</returns>
-        public static IServiceCollection AddDal(this IServiceCollection services, Uri cosmosUrl, string cosmosKey, string cosmosDatabase, string cosmosCollection)
+        public static IServiceCollection AddDal(this IServiceCollection services, Uri cosmosUrl, string cosmosKey, string cosmosDatabase, string cosmosCollection, int timeout, int retry)
         {
+            if (cosmosUrl == null)
+            {
+                throw new ApplicationException("cosmosUrl cannot be null");
+            }
+
+            CosmosClientOptions options = new CosmosClientOptions
+            {
+                RequestTimeout = TimeSpan.FromSeconds(timeout),
+                MaxRetryAttemptsOnRateLimitedRequests = retry,
+                MaxRetryWaitTimeOnRateLimitedRequests = TimeSpan.FromSeconds(timeout),
+            };
+
+            services.AddSingleton<CosmosClient>(new CosmosClient(cosmosUrl.AbsoluteUri, cosmosKey, options));
```

Comment:
I would build an extension so that you can call services.AddCosmosClient() or something similar - for ease of use and consistency

Reply:
added a new extension class and file